src/parquet/file/reader-internal.h

```diff
@@ -103,8 +103,8 @@ class SerializedFile : public ParquetFileReader::Contents {
   // This class does _not_ take ownership of the data source. You must manage its
   // lifetime separately
   static std::unique_ptr<ParquetFileReader::Contents> Open(
-      std::unique_ptr<RandomAccessSource> source,
-      ReaderProperties props = default_reader_properties());
+      const std::shared_ptr<RandomAccessSource>& source,
```

Comment:
Reverted the change, I agree that changing the pointer type in this patch is out of scope